### PR TITLE
Update Dockerfile

### PR DESCRIPTION
**Pull request by hmtai**
use multi stage to build image

https://github.com/kubeflow/pytorch-operator/issues/195

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,17 @@
+FROM golang:latest AS build-image
```
Comment (gaocegege) [inline Dockerfile]: I think we should avoid using latest image.

**Comment by gaocegege** [comment]
/ok-to-test

Thanks for your contribution! :tada: :+1:

Please sign the CLA, thanks.

**Comment by gaocegege** [comment]
Error logs of CI is here: https://paste.ubuntu.com/p/JwCfpc9vdQ/

File: Dockerfile (hunk 1 above)
Comment (hmtai) [reply Dockerfile]: Yes,use go version 1.12.

**Comment by johnugeorge** [comment]
@hmtai Docker file is called from https://github.com/kubeflow/pytorch-operator/blob/master/scripts/build.sh#L44

**Comment by hmtai** [comment]
@googlebot I signed it!

**Comment by hmtai** [comment]
@googlebot I signed it!)

**Comment by hmtai** [comment]
Close it and create PR 198.

**Pull request closed without merging**